### PR TITLE
Added ViewHistory List

### PR DESCRIPTION
**Pull request by Harineko0**
## Summary
- Modify heading of List.tsx
- Removed unnecessary default value settings and added setting viewHistory when updated. Also, change not to navigate to root when updated.
- Change viewHistoryApi to use useQueryClient() instead of import queryClient
- Added DocumentListSection to List.tsx
- Modify indent of UserItem.tsx
- Remove console.log from DocumentItem.
- Add limit:10 to userServiceImpl#searchUsers()
- Change userRepositoryImpl.test.ts to use toBeDefined()
- Change DocumentServiceImpl to set viewHistory when document is created or deleted.
- Change ViewHistoryService to allow passing document arguments.
- Fix: ViewHistoryRepositoryImpl updated ONLY updated_at so I fixed it
- Added ErrorContainer.tsx
- Installed shadcn/ui
- Made all repository and service injectable, added user exist and document exist assertion to ViewHistoryServiceImpl

## Summary by CodeRabbit

- **New Features**
  - Added configuration settings for a one-pager maker tool.
  - Added new dependencies for enhanced functionality and styling.

- **Bug Fixes**
  - Updated various API methods for improved document and user management.
  - Fixed issues with document and view history handling.

- **Refactor**
  - Refactored multiple service and repository methods for better code maintainability.
  - Updated method names and signatures for clarity and consistency.

- **Style**
  - Adjusted CSS selectors and custom properties for improved styling based on themes.

- **Tests**
  - Set up Firestore emulator for testing purposes.

- **Chores**
  - Removed unnecessary exclusion patterns and configurations.
  - Updated Redux store setup to remove deprecated reducers and middleware.

### CODE REVIEW

**Review by tako-t-t-1046 — APPROVED** [review]
Thanks a lot. LGTM👍

File: one-pager-maker/components.json
Comment (binnmti) [inline one-pager-maker/components.json:1]: What file is this for?

File: one-pager-maker/src/domain_service/userDomainService.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export interface UserDomainService {
-    isDuplicatedId(id: string): Promise<boolean>;
-}
\ No newline at end of file
+    exists(id: string): Promise<boolean>;
```
Comment (binnmti) [inline one-pager-maker/src/domain_service/userDomainService.ts:2]: good change

File: one-pager-maker/src/pages/List.tsx [hunk 2]
```diff
@@ -1,37 +1,33 @@
...
-    useFetchDocumentsQuery
-} from "../redux/document/documentsApi.ts";
 import {AddDocumentButton, DocumentItem} from "../stories/DocumentItem.tsx";
 import {useNavigate} from "react-router-dom";
 import {auth} from '../firebase';
+import {viewHistoryApi} from "../api/viewHistoryApi.ts";
+import {useMemo} from "react";
+import {documentApi} from "../api/documentApi.ts";
+import ErrorContainer from "@/stories/ErrorContainer.tsx";
+import {Document} from "@/entity/documentType.ts";
```
[the remark this answers is not in the document]
Comment (binnmti) [reply one-pager-maker/src/pages/List.tsx:8]: へー。そうなんだ

**Review by binnmti — APPROVED** [review]
LGTM

**Pull request merged**